src/screencast/screencast_common.c

```diff
@@ -0,0 +1,45 @@
+#include "screencast_common.h"
+
+char *strdup(const char *src) {
```

Comment:
`strdup` is a POSIX function. Add `#define _POSIX_C_SOURCE 200809L` and include `<string.h>` to get it.